assembly/index.d.ts

```diff
@@ -55,3 +61,48 @@ declare module "lunatic" {
     public drop(): void;
   }
 }
+
+declare module "net" {
+  /** A handle for a tcp socket connection. */
+  export class TCPSocket {
+    /** A method for connecting to a TCP server by it's IP address bytes. */
+    public static connect(ip: StaticArray<u8>, port: u16): TCPSocket | null;
+
+    /** A method for deserializing a TCPSocket across channels.  */
+    public static deserialize(value: u32): TCPSocket;
+
+    /** A method for serializing a tcp socket to be sent across channels. */
+    public serialize(): u32;
+
+    /** A method for reading a memory segment from the socket. Returns null */
+    public read(): StaticArray<u8> | null;
+
+    /** A method for reading socket data into an array of static arrays. Returns 0 if no bytes were read. */
```

Comment:
So this method actually blocks? We should document each function that blocks the main thread in the types so end users can make educated decisions on when to call them.

Reply:
It blocks from the developer's perspective, but not inside the runtime. I know this sounds a bit confusing :). It will block the process progress, but not actually the execution threads.

All calls in Lunatic are "blocking", but it's more like an `await` block in JS. Doing in JS
```javascript
const response = await fetch('url');
```
will "block" until the fetch finishes, but the thread is actually not blocked and other code can run in the meantime.

Lunatic does something similar, but without a need for the `await` keyword. It will just keep running other available processes on the operating system thread until the data is ready.

The developer should not be aware of this and the runtime will always do the most performant choice. That's why I don't think that we need to highlight this.

Reply:
I am still working out the wording. Just trying to get CI to work first.